src/postamble.js

```diff
@@ -188,19 +188,11 @@ Module['callMain'] = function callMain(args) {
   ensureInitRuntime();
 
   var argc = args.length+1;
-  function pad() {
-    for (var i = 0; i < {{{ QUANTUM_SIZE }}}-1; i++) {
-      argv.push(0);
-    }
-  }
-  var argv = [allocate(intArrayFromString(Module['thisProgram']), 'i8', ALLOC_NORMAL) ];
-  pad();
-  for (var i = 0; i < argc-1; i = i + 1) {
-    argv.push(allocate(intArrayFromString(args[i]), 'i8', ALLOC_NORMAL));
-    pad();
+  var argv = _malloc(argc * {{{ Runtime.POINTER_SIZE }}});
+  HEAP32[argv >> 2] = allocateUTF8(Module['thisProgram']);
+  for (var i = 0; i < argc-1; i++) {
+    HEAP32[(argv >> 2) + 1 + i] = allocateUTF8(args[i]);
   }
-  argv.push(0);
```

Comment:
Is `argv` still null-terminated in this PR?

Reply:
It's not, but I think that was never necessary, I can't find anything about it in C/C++ docs.

Reply:
Posts says it must be, although I have no access to actual C specification documents.

http://en.cppreference.com/w/cpp/language/main_function

>The value of argv[argc] is guaranteed to be a null pointer.

https://stackoverflow.com/questions/16418932/is-argvargc-equal-to-null-pointer

>The C Standard `5.1.2.2.1/2` second mark says explicitly

Reply:
Thanks! I missed that. Fixed.